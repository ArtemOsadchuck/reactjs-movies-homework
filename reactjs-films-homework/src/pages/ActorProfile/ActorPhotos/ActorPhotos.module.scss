$wite-color: rgb(255, 255, 255);

.PhotosWrapper {
  align-items: flex-end;
  color: $wite-color;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding-right: 0.5rem;
  padding-top: 1rem;
  width: 100%;

  .photosTitle {
    text-align: start;
    width: 68%;
  }

  .photosGrid {
    column-gap: 1rem;
    display: flex;
    margin-top: 1rem;
    min-height: 15rem;
    width: 68%;
  }

  .photo {
    border-radius: 5px;
    height: 11rem;
  }

  @media screen and (max-width: 1200px) {
    flex-wrap: wrap;
    padding: 1rem 1.5rem;
    width: 100%;

    .photosTitle {
      text-align: start;
      width: 100%;
    }

    .photosGrid {
      flex-wrap: wrap;
      justify-content: center;
      row-gap: 1rem;
      width: 100%;
    }
  }
}
