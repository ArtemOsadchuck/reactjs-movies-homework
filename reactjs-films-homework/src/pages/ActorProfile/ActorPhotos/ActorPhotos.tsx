import React from 'react';
import styles from './ActorPhotos.module.scss';

import getLang from '../../../languages/getLanguage';
import { useAppSelector } from '../../../hooks/hooks';

import { partOfImagesURL } from '../../../constants/links';

export interface IPhotos {
  aspect_ratio: number;
  file_path: string;
  height: number;
  iso_639_1: null;
  vote_average: number;
  vote_count: number;
  width: number;
}
export interface IActorPhotos {
  photos?: IPhotos[];
  photosLength?: number;
  nameAltImg: string | undefined;
}

const ActorPhotos: React.FC<IActorPhotos> = ({
  photos,
  photosLength,
  nameAltImg,
}) => {
  const appLang = useAppSelector((state) => state.mainReducer.lang);
  const titlePhotos = getLang(appLang).photos;
  console.log(photos);
  return (
    <div className={styles.photosWrapper}>
      {photos?.length ? (
        <h3 className={styles.photosTitle}>{titlePhotos}</h3>
      ) : null}
      <div className={styles.photosGrid}>
        {photos?.length
          ? photos.slice(0, photosLength).map((el) => {
              return (
                <img
                  width="120px"
                  className={styles.photo}
                  key={el.file_path}
                  src={`${partOfImagesURL}${el.file_path}`}
                  alt={nameAltImg}
                />
              );
            })
          : null}
      </div>
    </div>
  );
};

export default ActorPhotos;
