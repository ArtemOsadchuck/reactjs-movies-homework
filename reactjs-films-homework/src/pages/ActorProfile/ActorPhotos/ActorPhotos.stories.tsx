import React from 'react';
import { Meta, Story } from '@storybook/react';

import ActorPhotos, { IActorPhotos } from './ActorPhotos';
import ActorPhotosMocks from './mocks';

import { Provider } from 'react-redux';
import store from '../../../store/store';

const Template: Story<IActorPhotos> = (args) => {
  return (
    <Provider store={store}>
      <ActorPhotos photos={ActorPhotosMocks} photosLength={4} {...args} />
    </Provider>
  );
};

export const ActorPhotosGrig = Template.bind({});
ActorPhotosGrig.args = {
  photos: ActorPhotosMocks,
  photosLength: 4,
};

export default {
  title: 'ActorPhotos',
  component: ActorPhotos,
  argTypes: {
    photos: ActorPhotosMocks,
    photosLength: {
      control: { type: 'number' },
    },
  },
} as Meta;
