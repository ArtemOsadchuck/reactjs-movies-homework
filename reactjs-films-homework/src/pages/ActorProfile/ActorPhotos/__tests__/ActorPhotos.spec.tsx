import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';

import ActorPhotos, { IActorPhotos } from '../ActorPhotos';
import ActorPhotosMocks from '../mocks';

import RouterWrapper from '../../../../__testsUtils__/routerHoc';
import StoreWrapper from '../../../../__testsUtils__/storeHoc';

const renderComponent = ({
  photos,
  nameAltImg,
  photosLength,
}: IActorPhotos) => {
  const urlApp = '/movie-details';
  const { asFragment } = render(
    <StoreWrapper>
      <RouterWrapper url={urlApp}>
        <ActorPhotos
          photos={photos}
          nameAltImg={nameAltImg}
          photosLength={photosLength}
        />
      </RouterWrapper>
    </StoreWrapper>
  );
  return asFragment();
};

describe('ActorPhotos', () => {
  it('should renders correctly', () => {
    const mock = {
      nameAltImg: 'altName',
      photosLength: 4,
      photos: ActorPhotosMocks,
    };
    const fragment = renderComponent(mock);
    const imgTag = screen.getAllByRole('img').length;

    expect(fragment).toMatchSnapshot();
    expect(imgTag).toEqual(mock.photosLength);
  });

  cleanup();
});
