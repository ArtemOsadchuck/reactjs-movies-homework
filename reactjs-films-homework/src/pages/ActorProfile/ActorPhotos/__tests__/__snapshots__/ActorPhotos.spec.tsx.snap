// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorPhotos should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="photosWrapper"
  >
    <h3
      class="photosTitle"
    >
      Photos
    </h3>
    <div
      class="photosGrid"
    >
      <img
        alt="altName"
        class="photo"
        src="https://image.tmdb.org/t/p/w500//oIciQWr8VwKoR8TmAw1owaiZFyb.jpg"
        width="120px"
      />
      <img
        alt="altName"
        class="photo"
        src="https://image.tmdb.org/t/p/w500//1ahENoyEgKSbRUd0IsydIff7rt1.jpg"
        width="120px"
      />
      <img
        alt="altName"
        class="photo"
        src="https://image.tmdb.org/t/p/w500//20DXKyUQoL38SYoSW3JkHkZkkVP.jpg"
        width="120px"
      />
      <img
        alt="altName"
        class="photo"
        src="https://image.tmdb.org/t/p/w500//g2n87hnxD0w2suSBMv0BMoB4gef.jpg"
        width="120px"
      />
    </div>
  </div>
</DocumentFragment>
`;
